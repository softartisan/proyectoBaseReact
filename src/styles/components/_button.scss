.button{
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  text-decoration: none;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
}

.button--link{
  background: none;
}


.button--secondary{
  background: grey;
}