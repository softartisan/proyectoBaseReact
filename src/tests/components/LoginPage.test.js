import {shallow } from 'enzyme';
import React from 'react';
import {LoginPage} from '../../components/LoginPage';


test('Should render LoginPage', () => {
  const wrapper = shallow(<LoginPage/>);
  expect(wrapper).toMatchSnapshot();
})


test('Should call startLogin on button click', () => {
  const startLogin = jest.fn();
  const wrapper = shallow( <LoginPage startLogin={startLogin} /> );
  wrapper.find('button').at(0).simulate('click');
  expect(startLogin).toHaveBeenCalled();
})