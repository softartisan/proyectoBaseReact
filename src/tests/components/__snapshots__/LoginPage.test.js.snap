// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="box-layout__box"
      >
        <h1
          className="box-layout__title"
        >
          BoilerPlate
        </h1>
        <p>
          tagline for app.
        </p>
        <button
          className="button"
        >
          Iniciar sesión con Google
        </button>
      </div>,
      "className": "box-layout",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="box-layout__title"
          >
            BoilerPlate
          </h1>,
          <p>
            tagline for app.
          </p>,
          <button
            className="button"
          >
            Iniciar sesión con Google
          </button>,
        ],
        "className": "box-layout__box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BoilerPlate",
            "className": "box-layout__title",
          },
          "ref": null,
          "rendered": "BoilerPlate",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "tagline for app.",
          },
          "ref": null,
          "rendered": "tagline for app.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Iniciar sesión con Google",
            "className": "button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Iniciar sesión con Google",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="box-layout__box"
        >
          <h1
            className="box-layout__title"
          >
            BoilerPlate
          </h1>
          <p>
            tagline for app.
          </p>
          <button
            className="button"
          >
            Iniciar sesión con Google
          </button>
        </div>,
        "className": "box-layout",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="box-layout__title"
            >
              BoilerPlate
            </h1>,
            <p>
              tagline for app.
            </p>,
            <button
              className="button"
            >
              Iniciar sesión con Google
            </button>,
          ],
          "className": "box-layout__box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "BoilerPlate",
              "className": "box-layout__title",
            },
            "ref": null,
            "rendered": "BoilerPlate",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "tagline for app.",
            },
            "ref": null,
            "rendered": "tagline for app.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Iniciar sesión con Google",
              "className": "button",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Iniciar sesión con Google",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
